tests/test_types_typeddict.py

```diff
@@ -711,6 +711,63 @@ class RecursiveGenTypedDict(TypedDict, Generic[T]):
     ]
 
 
+def test_nested_generic_typeddict():
+    T = TypeVar('T')
+    U = TypeVar('U')
+
+    class BaseGenericTypedDict(TypedDict, Generic[U]):
+        format: U
+
+    class NestedGenericTypedDict(BaseGenericTypedDict[T]):
+        value: T
+
+    adapter = TypeAdapter(NestedGenericTypedDict[str])
```

Comment:
I feel like checking just one layer isn't enough here. The following adjustment to the test breaks:

```python
    V = TypeVar('V')

    class SubNestedGenericTypedDict(NestedGenericTypedDict[V]):
        info: V

    adapter = TypeAdapter(SubNestedGenericTypedDict[str])
```

... and then `fields['format']` again has schema `'any'`.

Reply:
Turns out this is because `mro(typed_dict_cls)` doesn't receive `BaseGenericTypedDict` 🫠
Looking into how to resolve this